cluster/src/main/scala/org/apache/spark/sql/streaming/SnappyStreamingContextFactory.scala

```diff
@@ -61,13 +56,18 @@ class SnappyStreamingContextFactory extends SparkContextFactory {
     new SnappyStreamingContext(LeadImpl.getInitializingSparkContext,
       Milliseconds(interval)) with ContextLike {
 
+      private val addedJars = scala.collection.mutable.ArrayBuffer.empty[String]
```

Comment:
@sumwale I checked the code. This is valid only for streaming context. The jars being added here only concern when we submit a streaming jobs. Hence one streaming context. That means it is just a object level attribute which is set when the  context is created. There is no other code path where this jars are added. 
Other context will have its own list. Ultimately one context will fail saying only one streaming contetxt is allowed.  Hence we can do away with the sync block.